src/main/java/mate/academy/controller/UserController.java

```diff
@@ -0,0 +1,24 @@
+package mate.academy.controller;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.user.UserResponseDto;
+import mate.academy.service.user.UserService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserService userService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @GetMapping("/find-all")
```

Comment:
`/find-all` doesn't follow RESTful best practices as it includes an action (find-all) which is more RPC-like (Remote Procedure Call) rather than resource-oriented. In REST the action should be implied by the HTTP method, and the URL should represent the resource - GET /users would be fine
```suggestion
    @GetMapping
```